packages/opentelemetry-sdk-trace-base/src/Span.ts

```diff
@@ -84,7 +97,16 @@ export class Span implements api.Span, ReadableSpan {
     this.parentSpanId = parentSpanId;
     this.kind = kind;
     this.links = links;
-    this.startTime = timeInputToHrTime(startTime ?? clock.now());
+
+    this._performanceStartTime = otperformance.now();
+    this._performanceOffset = Date.now() - this._performanceStartTime + otperformance.timeOrigin;
```

Comment:
This currently leads to `this._performanceOffset = 2 * performance.timeOrigin` in non-drift situations due to missing parenthesis

```suggestion
    this._performanceOffset = Date.now() - (this._performanceStartTime + otperformance.timeOrigin);
```

Reply:
I'm not sure if I'm just missing something obvious here but can you explain how? The order of `-` and `+` shouldn't matter and there are no other operations here.

Reply:
Simplified version: page loaded at 5000 (= performance.timeOrigin), it's been 2000 since then (= performance.now), and if current absolute time is at 7000, the current expected drift/offset to be 0

`this._performanceOffset = 7000 - 2000 + 5000 // = 10000, or into the future!`

Or to use right now values, welcome to 2075

![image](https://user-images.githubusercontent.com/1122555/201753608-f081eff2-c2fb-4abb-afea-e01bf2547524.png)


also looking at it again, these values are in milliseconds while they're later used as nanoseconds in addHrTimes (lines 224 & 228)

Reply:
ah I see it thanks

Reply:
>also looking at it again, these values are in milliseconds while they're later used as nanoseconds in addHrTimes (lines 224 & 228)

This was the root of the 0-duration spans in your example too